### DISCUSSION TITLE
Output 2 values

### DISCUSSION THREAD

**Discussion by AmriDomas** [teamalgoritma/community#749] [Q&A] [answered]
Dear Team Algoritma,

Berikut permasalahan yang saya dapatkan

<img width="485" alt="image" src="https://github.com/teamalgoritma/community/assets/130373441/3f6e2107-0025-4dad-b120-06525d30cbbe">

output yang saya dapatkan adalah 2, padahal yang saya cari 1 output, sehingga dalam melakukan evaluasi tidak bisa karena level berbeda.

<img width="289" alt="image" src="https://github.com/teamalgoritma/community/assets/130373441/39acfd5b-86d6-4dbe-8d95-b5b4d57e6acb">

<img width="384" alt="image" src="https://github.com/teamalgoritma/community/assets/130373441/5a73cc94-86d9-4516-912a-b1dfde7cfb81">

<img width="556" alt="image" src="https://github.com/teamalgoritma/community/assets/130373441/51f35193-ff62-4cde-9c71-339c477f0300">

Untuk preprocessing sebagai berikut:

<img width="374" alt="image" src="https://github.com/teamalgoritma/community/assets/130373441/833ba23e-dcfd-4b85-b18f-5a96b79eaaab">

<img width="302" alt="image" src="https://github.com/teamalgoritma/community/assets/130373441/3d31d17a-68e3-4c61-a5ac-3ab2c084f99b">

<img width="605" alt="image" src="https://github.com/teamalgoritma/community/assets/130373441/2df7adcb-993c-4823-886f-4887fa86ab0a">

<img width="639" alt="image" src="https://github.com/teamalgoritma/community/assets/130373441/9dc97c69-8e7c-4dc2-9370-d832af99e64a">

Output dalam preprocesing hanya 1 coloumn

Regards,
Amri

**Comment by kinantytasyao** [comment]
Dear Pak @AmriDomas 

Hal tersebut dikarenakan target variabelnya (Attrition) dalam tipe data factor, silakan dicoba kolom Attrition nya dijadikan integer dulu yaitu 0 dan 1. Baru dijalankan kembali model dan fungsi compute nya

Demikian yang dapat kami sampaikan, jika membantu boleh ditandai mark as answer. Terima kasih :)

**Reply by AmriDomas** [reply]
Ok. Terimakasih

**Comment by AmriDomas** [comment]
Maaf boleh bertanya lagi, ada permasalahan di bawah ini, pada saat di proses di r hasilnya berbeda pada saat setelah di kinit,
terlihat di bawah ini pada saat confusionmatrix di r dan di knit berbeda.

<img width="886" alt="image" src="https://github.com/teamalgoritma/community/assets/130373441/ac73d814-603e-40bc-baa2-3c9557a0da73">

untuk solusinya bagaimana?

Salam,
Amri

**Comment by Yusraf002** [comment, marked as answer]
Dear @AmriDomas 

Hal tersebut bisa terjadi karena proses randomizenya berbeda, hal ini bisa diatasi dengan melakukan `set.seed `pada saat  juga splitting data dan pembuatan model.

Demikian yang dapat kami sampaikan, jika membantu boleh ditandai mark as answer. Terima kasih :)

**Reply by AmriDomas** [reply]
Dear Team Algoritma,

Berikut langkah yang sudah saya terapkan

<img width="378" alt="image" src="https://github.com/teamalgoritma/community/assets/130373441/029d028e-05e7-4e7f-937b-7ce1eb158182">

set seed di spliting

<img width="293" alt="image" src="https://github.com/teamalgoritma/community/assets/130373441/55a56ab8-d942-457f-9a9f-2acaf1546f73">

set seed di model

<img width="344" alt="image" src="https://github.com/teamalgoritma/community/assets/130373441/6c212c66-43c8-49b4-8f2a-4ecc0bbc7e68">


set seed di  compile model

<img width="345" alt="image" src="https://github.com/teamalgoritma/community/assets/130373441/0c99762e-80d6-4768-aa19-12d27d24f2b5">

set seed di training model

<img width="859" alt="image" src="https://github.com/teamalgoritma/community/assets/130373441/846790c1-aaca-463d-bc83-f31a1c6b0360">

Hasilnya tetap saja tidak berubah setelah di knit, padahal di console pun juga berubah.

Mohon di berikan solusinya

Salam,
Amri

**Comment by Yusraf002** [comment]
Dear @AmriDomas 

sebelumnya apakah sudah dicoba dengan mengatur nilai `set.seed` yang sama untuk semua tahapan?

**Reply by AmriDomas** [reply]
Dear Team Algoritma,

set.seed sudah di atur semua dengan nilai yang sama, jika output prediksinya hanya berisi 1 nilai saja misal (0) maka nilai di knit dan di r sama, jika yang di prediksi nilai (1 dan 0) maka nilai setelah di knit berbeda dengan yang di r dan console
